commit bdfc33c176e2e3ac0ffb857d912dbe51f260646c
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:19 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/lamarck/patch-no-arch-flag.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 41e12fc13..2ac80129c 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -725,6 +725,10 @@ proc portconfigure::configure_get_universal_archflags {} {
 
 # internal proc to determine if the compiler supports -arch
 proc portconfigure::arch_flag_supported {compiler {multiple_arch_flags no}} {
+    global os.platform
+    if {${os.platform} ne "darwin"} {
+        return no
+    }
     if {${multiple_arch_flags}} {
         return [regexp {^gcc-4|llvm|apple|clang} ${compiler}]
     } else {
